spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/InferShape.scala

```diff
@@ -46,16 +45,20 @@ trait InferShape {
   // scalastyle:on
 
   /**
-   * We suppose the first dim is batch
+   * Return the inputShape for the current Layer and the first dim is batch.
    */
-  private[bigdl] final def getInputShape(): Shape = {
+  final def getInputShape(): Shape = {
+    require(this.isKerasStyle(),
+      "Torch style definition doesn't support getInputShape for now.")
     _inputShapeValue
   }
 
   /**
-   * We suppose the first dim is batch
+   * Return the outputShape for the current Layer and the first dim is batch.
```

Comment:
Can it return the output shape of a layer before it being added into a container?
It seems calling `Dense(10, inputShape=Shape(5)).getOutputShape()` will give null.